wsmaster/che-core-api-devfile/src/main/resources/build_docs.sh

```diff
@@ -62,23 +62,20 @@ build_native() {
 
 deploy() {
    cd ${TMP_DIR}
-   BRANCH_ARG=""
-   if [[ ! -z "${DEPLOY_BRANCH}" ]]; then
-       BRANCH_ARG="-b ${DEPLOY_BRANCH} --single-branch"
-   fi
-
    if [[ "$USE_SSH" == "true" ]]; then
      DOCS_REPOSITORY_URL="git@github.com:${DEPLOY_ORGANIZATION}/devfile.git"
    else
      DOCS_REPOSITORY_URL=https://${GH_TOKEN}@github.com/${DEPLOY_ORGANIZATION}/devfile.git
    fi
 
-   rm -rf devfile && git clone ${BRANCH_ARG} ${DOCS_REPOSITORY_URL}
+   rm -rf devfile && git clone ${DOCS_REPOSITORY_URL}
    cp -f docs/* ./devfile/docs
    cd devfile
    if [[ `git status --porcelain` ]]; then
+       git checkout -b "${DEPLOY_BRANCH}"
```

Comment:
What if `DOCS_REPOSITORY` already has such a branch?
I assume that error will happen during push if the remote branch is not based on top of master, right?

Reply:
i think checkout should fail saying "branch already exists" ?

Reply:
Please check
It's what I see
![Screenshot_20190314_095702](https://user-images.githubusercontent.com/5887312/54340196-8cc33c80-463f-11e9-9730-1cb7cdd0b53b.png)

Reply:
so you trying tu push local branch to another remote, right ?

Reply:
I tried to push the local branch (is created from the default branch) to a remote branch with the same name (remote is not up to date to master).

Reply:
so what is the solution ?

Reply:
I see three options here:
1. BASE_BRANCH parameter. See https://github.com/eclipse/che/pull/12880#discussion_r265450066.
2. force push (it is dangerous).
3. rework logic and checkout to existing remote branch or create a new if a remote branch does not exist.

Reply:
I don't see how the base branch helps, it will be always "master" on the CI. And applying more changes into existing remote branch... also questionable for me.

Reply:
i think the best way is to fail when branch exists on the remote. It means that previous run was unfinished (i.e PR not created & merged, or branch not cleared after that)